e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/HasuraRequests.java

```diff
@@ -588,6 +608,38 @@ public SchedulingResponse awaitScheduling(int schedulingSpecId, int timeout) thr
     throw new TimeoutError("Scheduling timed out after " + timeout + " seconds");
   }
 
+  /**
+   * Run scheduling on the specified scheduling specification with a timeout of 30 seconds
+   * Used when the scheduling run is expected to fail.
+   */
+  public SchedulingResponse awaitFailingScheduling(int schedulingSpecId) throws IOException {
+    return awaitFailingScheduling(schedulingSpecId, 30);
+  }
+
+  /**
+   * Run scheduling on the specified scheduling specification
+   * Used when the scheduling run is expected to fail.
+   */
+  public SchedulingResponse awaitFailingScheduling(int schedulingSpecId, int timeout) throws IOException {
+    for(int i = 0; i < timeout; ++i){
+      final var response = schedule(schedulingSpecId);
+      switch (response.status()) {
+        case "pending", "incomplete" -> {
+          try {
+            Thread.sleep(1000); // 1s
+          } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        }
+        case "complete", "failed" -> {
```

Comment:
`case "complete"` should be `fail()` or caught by `default` case -- if the scheduling request didn't fail, then something's wrong.

Reply:
done, good catch